### PR TITLE
Allow pass shop to omniauth strategy as param

### PR DESCRIPTION
**Pull request by ildarkayumov**
Based on #72 

I need to use [omniauth-shopify-oauth2](https://github.com/Shopify/omniauth-shopify-oauth2) without [shopify_app](https://github.com/Shopify/shopify_app). And it looks like it requires custom client option [setup](https://github.com/Shopify/omniauth-shopify-oauth2/blob/master/lib/omniauth/strategies/shopify.rb#L26)
Another option is to duplicate shopify_app controller sessions#login

This pr allows to get shop name from params if it hasn't been provided with `session['shopify.omniauth_params']`

### CODE REVIEW

File: lib/omniauth/strategies/shopify.rb [hunk 1]
```diff
@@ -26,7 +26,11 @@ class Shopify < OmniAuth::Strategies::OAuth2
       option :setup, proc { |env|
         strategy = env['omniauth.strategy']
 
-        shopify_auth_params = strategy.session['shopify.omniauth_params'] && strategy.session['shopify.omniauth_params'].with_indifferent_access
+        shopify_auth_params = strategy.session['shopify.omniauth_params'] ||
+          strategy.session['omniauth.params'] ||
```
Comment (tanema) [inline lib/omniauth/strategies/shopify.rb:30]: Why is `with_indifferent_access` removed here?

File: example/config.ru [hunk 2]
```diff
@@ -1,5 +1,6 @@
 require 'bundler/setup'
 require 'sinatra/base'
+require 'active_support/core_ext/hash'
```
Comment (tanema) [inline example/config.ru:3]: Why was this change needed?

File: lib/omniauth/strategies/shopify.rb (hunk 1 above)
Comment (ildarkayumov) [reply lib/omniauth/strategies/shopify.rb:30]: moved here https://github.com/Shopify/omniauth-shopify-oauth2/pull/91/files#diff-20fb14b92659da69e72effc9ab8f72feR33 to prevent duplicated call `with_indifferent_access` for each of "params place"

I found only 3 places `shop` param come from:
`strategy.session['shopify.omniauth_params']` - set by shopify app gem
`strategy.session['omniauth.params']` - omniauth sets this key in request phase(https://github.com/omniauth/omniauth/blob/master/lib/omniauth/strategy.rb#L208) and deletes it before callback. But setup phase runs before delete. So we can use this key on callback phase
`strategy.request.params` - get shop from params

File: example/config.ru (hunk 2 above)
Comment (ildarkayumov) [reply example/config.ru:3]: because it raises an error `undefined method 'with_indifferent_access' for #<Hash:0x1012392c0>` I don't see any other filed(except integration_test.rb) `active_support/core_ext/hash` has been required.

**Comment by ildarkayumov** [comment]
sure 👍
